packages/sdk-codegen/src/sdkModels.spec.ts

```diff
@@ -366,6 +366,48 @@ describe('sdkModels', () => {
       ])
     })
 
+    it('Setting', () => {
```

Comment:
may I suggest changing the test names to be more descriptive? This test file is quite big and it would make it easier in case anyone needs to venture here again. Feel free to ignore these suggestions, the tests been fine as is but in my experience it's sometimes difficult to find what applies if you don't know what you're looking for

```suggestion
    it('generates writeable type for nested types with some readonly properties', () => {
```